test/Test/IO/Tinfoil/Signing/Ed25519/Internal.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE GADTs #-}
+{-# OPTIONS_GHC -fno-warn-missing-signatures #-}
```

Comment:
Since you actually have signatures for everything, maybe worth dropping `-fno-warn-missing-signatures`.